www/src/js/index.js

```diff
@@ -61,36 +61,44 @@ function setActiveToc() {
     // if (window.innerWidth >= 860) {
     //   tocEl.scrollIntoView({behavior: 'smooth'});
     // }
-      //   {
-      //   top:
-      //     tocEl.getBoundingClientRect().top + gridTocEl.scrollTop - PADDING_TOP,
-      //   behavior: 'smooth',
-      // });
+    //   {
+    //   top:
+    //     tocEl.getBoundingClientRect().top + gridTocEl.scrollTop - PADDING_TOP,
+    //   behavior: 'smooth',
+    // });
     return;
   }
 }
 
 const gridBodyEl = document.getElementById('grid-body');
 const tableOfContentsEl = document.querySelector('.sub-navigation .toc');
-const gridTocEl = document.querySelector('.grid-toc');
+const gridTocEl = document.querySelector('#nav-primary');
 gridBodyEl.addEventListener('scroll', debounce(setActiveToc));
 window.addEventListener('scroll', debounce(setActiveToc));
 
-document.getElementById('toc-drawer-button').addEventListener('click', (e) => {
-  e.preventDefault();
+function handleMobileNav(evt) {
+  evt.preventDefault();
   /*If hidden-mobile class is enabled that means we are on desktop do overflow normal but we
     if we are at mobile fixed body position, so that its not scrollable(which currently causing bug) and navbar  handling its
     owns scroll. Case to consider there are chance use can open navbar using toggle button and user when click on any link
     body postion should be unset
     */
-  const ishiddenMobileClassEnabled = gridTocEl.classList.toggle(
-    'hidden-mobile',
-  );
-});
-gridTocEl.addEventListener('click', (e) => {
-  gridTocEl.classList.add('hidden-mobile');
-  document.body.style.position = '';
-});
+  gridTocEl.classList.toggle('is-visible');
+
+  const isOpen = gridTocEl.classList.contains('is-visible');
+  if (isOpen) {
+    evt.target.setAttribute('aria-expanded', 'true');
+    document.body.style.overflow = 'hidden';
```

Comment:
We had some weird mobile scroll behavior, where the page could still scroll behind the nav when open. This fixes that.